controller/backing_image_controller.go

```diff
@@ -310,22 +345,530 @@ func (bic *BackingImageController) syncBackingImage(key string) (err error) {
 
 	bic.cleanupEvictionRequestedBackingImageCopies(backingImage)
 
+	if types.IsDataEngineV2(backingImage.Spec.DataEngine) {
+		if err := bic.handleV2BackingImage(backingImage); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (bic *BackingImageController) handleV2BackingImage(bi *longhorn.BackingImage) (err error) {
+	if err := bic.prepareFirstV2Copy(bi); err != nil {
+		return errors.Wrapf(err, "failed to prepare the first v2 backing image")
+	}
+
+	if err := bic.syncV2StatusWithInstanceManager(bi); err != nil {
+		return errors.Wrapf(err, "failed to sync v2 backing image status from instance manager")
+	}
+
+	if err := bic.deleteInvalidV2Copy(bi); err != nil {
+		return errors.Wrapf(err, "failed to delete invalid v2 backing image")
+	}
+
+	if err := bic.prepareV2Copy(bi); err != nil {
+		return errors.Wrapf(err, "failed to prepare v2 backing image")
+	}
+
+	return nil
+}
+
+func (bic *BackingImageController) isFirstV2CopyInState(bi *longhorn.BackingImage, state longhorn.BackingImageState) bool {
+	if bi.Status.V2FirstCopyDisk != "" && bi.Status.V2FirstCopyStatus == state {
+		return true
+	}
+
+	if bi.Status.V2FirstCopyDisk != "" {
+		if status, exists := bi.Status.DiskFileStatusMap[bi.Status.V2FirstCopyDisk]; exists && status.State == state {
+			return true
+		}
+	}
+
+	return false
+}
+
+func (bic *BackingImageController) cleanupFirstFailedV2Copy(bi *longhorn.BackingImage) (err error) {
+	if bi.Status.V2FirstCopyDisk == "" {
+		return nil
+	}
+
+	log := getLoggerForBackingImage(bic.logger, bi)
+
+	node, _, err := bic.ds.GetReadyDiskNodeRO(bi.Status.V2FirstCopyDisk)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get the ready disk node for disk %v", bi.Status.V2FirstCopyDisk)
+	}
+
+	engineClientProxy, err := bic.getEngineClientProxy(node.Name, longhorn.DataEngineTypeV2, log)
+	if err != nil {
+		return errors.Wrapf(err, "failed to get the engine client proxy for node %v", node.Name)
+	}
+	defer engineClientProxy.Close()
+
+	// clean up the failed first v2 copy
+	err = engineClientProxy.SPDKBackingImageDelete(bi.Name, bi.Status.V2FirstCopyDisk)
+	if err != nil {
+		return errors.Wrapf(err, "failed to delete the failed v2 copy on the disk %v", bi.Status.V2FirstCopyDisk)
+	}
+	delete(bi.Spec.DiskFileSpecMap, bi.Status.V2FirstCopyDisk)
```

Comment:
ummm
delete the backing image in the spdk means
1. delete the "record" in the spdk server `s.BackingImageMap`
2. delete the snapshot lvol

I don't think it can fail simply
however if your are talking about network issue
Then 

1. delete the snapshot lvol in the spdk server
2. restart the instance manager so the "record" in the s.BackingImageMap is gone
3. manually edit the backing image CR to remove the spec and the status.